.github/workflows/scripts/prev_tag.sh

```diff
@@ -1,61 +1,134 @@
+#
+# Release Candidates (RC) for Major Versions:
+# - For RC releases of a major version (e.g., 9.0.0-rc1), the previous tag is the base (x.0) tag of the highest minor version from the last major series (e.g., 8.9.0).
+
+# Function to determine the previous tag based on the type and version
+determine_prev_tag() {
+  local current_tag=$1
+  local type=$2
+  local major_minor=$(echo "$current_tag" | cut -d '.' -f 1,2)
+  local patch_version=$(echo "$current_tag" | cut -d '.' -f 3 | grep -oE "^[0-9]+")
+  local prev_major_minor=$(echo "$major_minor" | awk -F. '{$NF = $NF - 1;} 1' | sed 's/ /./g')
+
+  local prev_tag=""
+  local prev_tags=$(git --no-pager tag --sort=-v:refname)
+
+case $type in
+  NORMAL)
+    # For a normal release, find the highest tag that is lower than the current tag in the same minor version series
+    prev_tag=$(echo "$prev_tags" | grep -E "^$major_minor\.[0-9]+$" | awk -v curTag="$current_tag" '$0 < curTag' | head -1)
+
+    # If no previous tag is found within the same minor version, default to the .0 tag of the previous minor version
+    [[ -z $prev_tag ]] && prev_tag="$prev_major_minor.0"
+    ;;
+
+  RC)
+    # Remove the RC part from the current tag to find the base version
+    local base_version=$(echo "$current_tag" | sed 's/-rc[0-9]*$//')
+      # For regular RC releases, find the highest RC tag that is lower than the current tag in the same minor version series
+      prev_tag=$(echo "$prev_tags" | grep -E "^$major_minor\.[0-9]+-rc[0-9]+$" | awk -v curTag="$current_tag" '$0 < curTag' | head -1)
+      # If no previous RC is found within the same minor version, default to the .0 tag of the previous minor version
+      [[ -z $prev_tag ]] && prev_tag="$prev_major_minor.0"
+    ;;
+
+ALPHA)
+  if [[ $current_tag =~ -alpha[0-9]+-rc[0-9]+$ ]]; then
+    local alpha_version=$(echo "$current_tag" | sed 's/-rc[0-9]*$//')
+    local base_alpha=$(echo "$alpha_version" | sed 's/-alpha[0-9]*$//')
+
+    # Get the list of all RCs for the current alpha version
+    local rc_tags=$(echo "$prev_tags" | grep -E "^$alpha_version-rc[0-9]+$" | sort -rV)
+
+    if [[ -z $rc_tags ]]; then
+      # If no RCs found for the current alpha version, set the previous tag to the highest alpha version before the current alpha
+      prev_tag=$(echo "$prev_tags" | grep -E "^$base_alpha-alpha[0-9]+$" | sort -rV | awk -v curAlpha="$alpha_version" '$0 < curAlpha' | head -1)
+    else
+      # Find the immediate preceding RC for the current tag
+      prev_tag=$(echo "$rc_tags" | awk -v curTag="$current_tag" '$0 < curTag' | head -1)
+      if [[ -z $prev_tag ]]; then
+        # If no preceding RC found, it means the current tag is the first RC of the alpha version
+        # So, set the previous tag to the highest alpha version before the current alpha
+        prev_tag=$(echo "$prev_tags" | grep -E "^$base_alpha-alpha[0-9]+$" | sort -rV | awk -v curAlpha="$alpha_version" '$0 < curAlpha' | head -1)
```

Comment:
It looks like the right side of this assignment is repeated on line 64.

Consider extracting this code into its own function named something like `highest_alpha_version_before()`